app/views/users/new.html.haml

```diff
@@ -14,6 +14,10 @@
             .col-sm-6
               = f.password_field :password, hide_label: true
           .form-group
+            %label.control-label.col-sm-2 Confirm Password
+            .col-sm-6
+              = f.password_field :password_confirmation, hide_label: true
```

Comment:
https://github.com/bootstrap-ruby/rails-bootstrap-forms#horizontal-forms

Take a look at that link to see how to adjust the widths on the form, as well as a proper example.